hawkular-inventory-bus/src/main/java/org/hawkular/inventory/bus/MessageSender.java

```diff
@@ -32,23 +37,26 @@
  * @since 0.0.1
  */
 final class MessageSender {
-    private final ProducerConnectionContext producerConnectionContext;
     private final MessageProcessor messageProcessor;
+    private final Configuration configuration;
 
-    public MessageSender(ProducerConnectionContext producerConnectionContext) {
-        this.producerConnectionContext = producerConnectionContext;
+
+    public MessageSender(Configuration configuration) {
+        this.configuration = configuration;
         this.messageProcessor = new MessageProcessor();
     }
 
     public void send(Interest<?, ?> interest, Object inventoryEvent) {
         InventoryEvent<?> message = InventoryEvent.from(interest.getAction(), inventoryEvent);
         try {
+            ProducerConnectionContext producerConnectionContext = init();
```

Comment:
Doesn't this leak connections or something? I'm not sure, just asking if re-initializing everything for each message sent is really what we want here.